autopilot/autopilot.go

```diff
@@ -209,23 +218,27 @@ func (ap *Autopilot) Run() error {
 			// reset forceScan
 			forceScan = false
 
-			// block until the autopilot is configured
-			if configured, interrupted := ap.blockUntilConfigured(ap.ticker.C); !configured {
+			// block until consensus is synced
+			if synced, blocked, interrupted := ap.blockUntilSynced(ap.ticker.C); !synced {
 				if interrupted {
 					close(tickerFired)
 					return
 				}
-				ap.logger.Error("autopilot stopped before it was able to confirm it was configured in the bus")
+				ap.logger.Error("autopilot stopped before consensus was synced")
 				return
+			} else if blocked {
```

Comment:
Seems unnecessary to return `blocked` in `blockUntilSynced`. Can't you check whether `ap.stopChan` was closed?

Reply:
Not sure how `ap.stopChan` would be at play here. What this achieves is a rescan in case we were blocking on consensus to get synced. E.g. we try and perform host scan but we are unaware of any hosts, we then go on to block and wait until we are synced. When that eventually unblocks, I figured it's nice to try and perform another host scan in the event the first scan completed.